pkg/storageos/scheduler_extender.go

```diff
@@ -0,0 +1,280 @@
+package storageos
+
+import (
+	"bytes"
+	"fmt"
+	"text/template"
+
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const (
+	policyConfigMapName          = "storageos-scheduler-policy"
+	policyConfigKey              = "policy.cfg"
+	schedulerConfigConfigMapName = "storageos-scheduler-config"
+	schedulerConfigKey           = "config.yaml"
+
+	// schedulerReplicas is the number of instances of kube-scheduler.
+	schedulerReplicas = 1
+)
+
+func (s *Deployment) createSchedulerExtender() error {
+	// Create configmap with scheduler configuration and policy.
+	if err := s.createSchedulerPolicy(); err != nil {
+		return err
+	}
+	if err := s.createSchedulerConfiguration(); err != nil {
+		return err
+	}
+
+	// Create RBAC related resources.
+	if err := s.createClusterRoleForScheduler(); err != nil {
+		return err
+	}
+	if err := s.createServiceAccountForScheduler(); err != nil {
+		return err
+	}
+	if err := s.createClusterRoleBindingForScheduler(); err != nil {
+		return err
+	}
```

Comment:
Creating all these associated components here to reduce cluttering `Deploy()`.
I'll change CSI helper deployment similar to this in a separate PR.